backend.rb

```diff
@@ -57,14 +56,13 @@ class Backend < Sinatra::Base
       origin_proof:  params[:origin_proof]
     )
 
-    authorization_response = purchase.authorize!
-    if authorization_response.success?
+    success, response_data, response_code = purchase.order!
+    if success
       @user.subscribe!
       article = Article.find(params[:article_id])
       return json data: article.paid_content
```

Comment:
please return the `klarna_response` also when the request succeeds (needed for the order_reference feature)